src/lib/Dungeon.js

```diff
@@ -441,19 +634,7 @@ class AutomationDungeon
 
                 if (DungeonRunner.map.hasAccessToTile(currentLocation))
                 {
-                    // Store chest positions
-                    if (tile.type() === GameConstants.DungeonTile.chest)
-                    {
-                        this.__internal__addChestPosition(currentLocation);
-                    }
-
-                    if (tile.type() === GameConstants.DungeonTile.boss)
-                    {
-                        // Only tag the dungeon as completed if it's not the first move or any tile is visited apart from the entrance
-                        this.__internal__isCompleted = (!this.__internal__isFirstMove) || DungeonRunner.map.board()[currentFloor].some(
-                            (row) => row.some((tile) => tile.isVisited && (tile.type() !== GameConstants.DungeonTile.entrance)));
-                        this.__internal__bossPosition = currentLocation;
-                    }
+                    this.__internal__markCell({ ...currentLocation, tile });
```

Comment:
any reason to do it only if the player has access to it in this case ? 
The other calls are not protected this way

Reply:
Looks good to me then